lib/BuildSystem/SwiftTools.cpp

```diff
@@ -348,37 +358,47 @@ class SwiftCompilerShellCommand : public ExternalCommand {
     }
 
     os << "{\n";
+
+    // Write out the entries for each source file if WMO is not enabled.
```

Comment:
I think there are cases where WMO still uses the per-file entries (with parallel WMO). I unfortunately forget the details, but there are multiple modes, one where WMO runs as a single process, and one where WMO runs with one thread/process to do the codegen of the .o files (I think there is an explicit option for it).

Xcode's default these days is to use the one which does the codegen in parallel, so we should match that. In that mode, I believe the individual .o files are still used.